.github/workflows/build.yaml

```diff
@@ -0,0 +1,38 @@
+name: build
+
+on:
+  push:
+    branches:
+      - master
+jobs:
+  build:
+    name: build
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Set up Go 1.17
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.8
```

Comment:
Should we use a matrix to test multiple go versions?
WDYT?

Reply:
sure
adding 16, 17 and 18 for now